.circleci/scripts/setup_linux_system_environment.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+set -ex
+
+# Set up CircleCI GPG keys for apt, if needed
+curl -L https://packagecloud.io/circleci/trusty/gpgkey | sudo apt-key add -
+
+# Stop background apt updates.  Hypothetically, the kill should not
+# be necessary, because stop is supposed to send a kill signal to
+# the process, but we've added it for good luck.  Also
+# hypothetically, it's supposed to be unnecessary to wait for
+# the process to block.  We also have that line for good luck.
+# If you like, try deleting them and seeing if it works.
+sudo systemctl stop apt-daily.service || true
+sudo systemctl kill --kill-who=all apt-daily.service || true
+
+sudo systemctl stop unattended-upgrades.service || true
+sudo systemctl kill --kill-who=all unattended-upgrades.service || true
+
+# wait until `apt-get update` has been killed
+while systemctl is-active --quiet apt-daily.service
+do
+    sleep 1;
+done
+while systemctl is-active --quiet unattended-upgrades.service
+do
+    sleep 1;
+done
+
+# See if we actually were successful
+systemctl list-units --all | cat
+
+sudo apt-get purge -y unattended-upgrades
+
+cat /etc/apt/sources.list
+
+# TODO: https://circleci.com/gh/pytorch/pytorch/1478441
+# The first `grep` here is exiting with exit code 1 when running in CI.
+# The output of `ps auxfww` does not match `[a]pt`.
+# ps auxfww | grep '[a]pt'
```

Comment:
@kostmo I'm having a very strange issue with this `grep`. When I extracted the script out to a file, this started to fail. `grep` exits with code 1 (no output matched).

Any ideas what's up?

Reply:
I think these are in there to try and detect an issue we were having with a stray `apt` process leftover from the VM provisionining.

You shouldn't be getting failures though, I can't see a `-o pipefail` anywhere.

Reply:
Pipefail might be inherited differently from the parent shell (the shell the build step is executed in) when invoking the script in this way.